catalyst/utils/config_validator.py

```diff
@@ -0,0 +1,92 @@
+from schema import Schema, And, Optional, Forbidden, SchemaError
+from catalyst.dl.registry import (
+    CALLBACKS, CRITERIONS, MODELS, OPTIMIZERS, SCHEDULERS
+)
+from catalyst.utils.registry import RegistryException
+
+TOP_LEVEL_PARAMS = {
+    'model_params': object,
+    'args': object,
+    'stages': object
+}
+
+MODEL_PARAMS = {
+    'model': And(str, len)
+}
+ARGS_PARAMS = {
+    'expdir': And(str, len),
+    Forbidden('logdir'): '.',
```

Comment:
This is pretty strange constraint, but you can't write logs in same directory because some file-copying fails.